opensearch/setup_models.py

```diff
@@ -0,0 +1,553 @@
+            "embedding_dimension": 384,
+            "framework_type": "huggingface_transformers",
+            "pooling_mode": "MEAN",
+            "normalize_result": True,
+            "all_config": '{"_name_or_path":"nreimers/MiniLM-L6-H384-uncased",\
+"architectures":["BertModel"],"attention_probs_dropout_prob":0.1,"gradient_checkpointing"\
+:false,"hidden_act":"gelu","hidden_dropout_prob":0.1,"hidden_size":384,"initializer_range"\
+:0.02,"intermediate_size":1536,"layer_norm_eps":1e-12,"max_position_embeddings":512,\
+"model_type":"bert","num_attention_heads":12,"num_hidden_layers":6,"pad_token_id":0,\
+"position_embedding_type":"absolute","transformers_version":"4.8.2","type_vocab_size"\
+:2,"use_cache":true,"vocab_size":30522}',
+        },
+        "created_time": 1676329221436,
+        "url": "https://artifacts.opensearch.org/models/ml-models/huggingface/sentence-\
+transformers/all-MiniLM-L6-v2/1.0.1/onnx/sentence-transformers_all-MiniLM-L6-v2-1.0.1-onnx.zip",
+    }
+    return setup_model(register_body, model_name)
+
+
+def setup_reranking_model(model_group_id):
+    """
+    Register and deploy bge-reranker-base-quantized
+    """
+    model_name = "reranking"
+    all_config = {
+        "_name_or_path": "BAAI/bge-reranker-base",
+        "architectures": ["XLMRobertaForSequenceClassification"],
+        "attention_probs_dropout_prob": 0.1,
+        "bos_token_id": 0,
+        "classifier_dropout": None,
+        "eos_token_id": 2,
+        "hidden_act": "gelu",
+        "hidden_dropout_prob": 0.1,
+        "hidden_size": 768,
+        "id2label": {"0": "LABEL_0"},
+        "initializer_range": 0.02,
+        "intermediate_size": 3072,
+        "label2id": {"LABEL_0": 0},
+        "layer_norm_eps": 1e-05,
+        "max_position_embeddings": 514,
+        "model_type": "xlm-roberta",
+        "num_attention_heads": 12,
+        "num_hidden_layers": 12,
+        "output_past": True,
+        "pad_token_id": 1,
+        "position_embedding_type": "absolute",
+        "torch_dtype": "float32",
+        "transformers_version": "4.33.3",
+        "type_vocab_size": 1,
+        "use_cache": True,
+        "vocab_size": 250002,
+    }
+    register_body = {
+        "name": "BAAI/bge-reranker-base-quantized",
+        "version": "1.0.0",
+        "description": "Cross Encoder text similarity model",
+        "model_format": "ONNX",
+        "function_name": "TEXT_SIMILARITY",
+        "model_group_id": model_group_id,
+        "model_content_hash_value": "04157d66d847d08b3d2b51ad36cf0e1fb82afadb8086212a1d2bac2b7d6fe08a",
+        "model_config": {
+            "model_type": "roberta",
+            "embedding_dimension": 1,
+            "framework_type": "huggingface_transformers",
+            "all_config": json.dumps(all_config),
+        },
+        "url": "https://aryn-public.s3.amazonaws.com/models/BAAI/bge-reranker-base-quantized-2.zip",
+    }
+    return setup_model(register_body, model_name, register_timeout=120)
```

Comment:
we can... It might make it easier to switch out our models at the cost of being able to understand what the hell is going on. Not sure that tradeoff is worth it